src/pycropml/pparse.py

```diff
@@ -111,11 +112,17 @@ def Algorithm(self, elt):
         print('Algorithm')
         
         language=elt.attrib["language"]
+        platform=elt.attrib["platform"]
         development = elt.text
         
-        algo = algorithm.Algorithm(language, development)
+        if "function" in elt.attrib.keys(): 
```

Comment:
Why using keys?
This is the natural behavior of dict.
Just use
    if "function" in elt.attrib: